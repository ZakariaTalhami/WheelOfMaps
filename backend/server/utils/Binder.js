/**
 * Binds all the methods within a class to "this"
 */
class Binder {}

/**
 * Get all the methods in the given instance
 * @param {Object} instance
 * @return {Array} - list of methods
 */
Binder.getAllMethods = function (instance) {
    var props = [];
    var obj = instance;
    while ((obj = Object.getPrototypeOf(obj)) && obj != Object.prototype) {
        props = props.concat(Object.getOwnPropertyNames(obj));
    }

    return props.sort().filter(function (e, i, arr) {
        if (e != arr[i + 1] && typeof instance[e] == "function") return true;
    });
};

/**
 * Binds the instance's methods
 * @param {Object} instance
 */
Binder.bind = function (instance) {
    Binder.getAllMethods(instance).forEach((mtd) => {
        instance[mtd] = instance[mtd].bind(instance);
    });
};

export default Binder;
