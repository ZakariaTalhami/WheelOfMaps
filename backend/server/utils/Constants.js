export const CHAPTER_IDNEX_RANGE_REGEX = /^\d{5}(-\d{5})?$/;
export const CHAPTER_RANGE_SCHEMA = {
    type: String,
    required: true,
    match: CHAPTER_IDNEX_RANGE_REGEX,
};
