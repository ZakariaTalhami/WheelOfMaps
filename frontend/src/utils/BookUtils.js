/*
    Uility module for common functionality for Books and chapters.
*/

export const getBookChapterTitles = (book) => {
    const chapters = book.chapters || [];
    return chapters.map((chapter) => {
        const prefix =
            chapter.number === 0 ? "Prologue" : `Chapter ${chapter.number}`;

        return `${prefix}: ${chapter.title}`;
    });
};
