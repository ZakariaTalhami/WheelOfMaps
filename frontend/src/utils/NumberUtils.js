/**
 * Pads a number depending on the number of digits required
 * Returns a string with the number padded with zeros
 *
 * @param {Number} num - the desired number
 * @param {Number} places - number of digits
 * @returns {String} formated string with zero padding
 */
export function zeroPad(num, places) {
    var zero = places - num.toString().length + 1;
    return Array(+(zero > 0 && zero)).join("0") + num;
}
