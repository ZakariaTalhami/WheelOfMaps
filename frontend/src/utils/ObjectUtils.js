import _ from "lodash";

const evaluateObjectPath = (obj, path = [], defaultValue) => {
    if (_.isString(path)) path = _.split(path, ".");
    if (!_.isObject(obj) || _.isEmpty(path)) return obj || defaultValue;

    return evaluateObjectPath(obj[_.head(path)], _.drop(path), defaultValue);
};

const ObjectUtils = {
    evaluateObjectPath,
};

export default ObjectUtils;
