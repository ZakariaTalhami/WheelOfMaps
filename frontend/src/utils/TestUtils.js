import React from "react";
import { render } from "@testing-library/react";
import { ChakraProvider } from "@chakra-ui/react";
//Global Theme
import extendedTheme from "../assets/styles/extendedTheme";
// TODO: remove custom font icons, replace with react-icons
import "../assets/fonts/style.css";
// Leafletjs Styles
import "leaflet/dist/leaflet.css";
import "../assets/_leafletjs.override.scss";

const AllTheProviders = ({ children }) => {
    Object.defineProperty(window, "matchMedia", {
        writable: true,
        value: jest.fn().mockImplementation((query) => ({
            matches: false,
            media: query,
            onchange: null,
            addListener: jest.fn(), // deprecated
            removeListener: jest.fn(), // deprecated
            addEventListener: jest.fn(),
            removeEventListener: jest.fn(),
            dispatchEvent: jest.fn(),
        })),
    });

    return <ChakraProvider theme={extendedTheme}>{children}</ChakraProvider>;
};

const customRender = (ui, options) =>
    render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
