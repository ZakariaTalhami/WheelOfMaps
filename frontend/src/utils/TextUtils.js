import { Text } from "@chakra-ui/react";

/**
 * Split the summary by new line and wrap in paragraphs
 * @param {String} summary
 * @returns {Array} paragraphs of the summary
 */
export const getSummaryParagraphs = (summary) => {
    const paragraphs = summary.split("\r\n\r\n");
    return paragraphs.map((para, index) => (
        <Text
            w="100%"
            role="paragraph"
            style={{ textIndent: "1.5rem" }}
            key={index}
        >
            {para}
        </Text>
    ));
};
