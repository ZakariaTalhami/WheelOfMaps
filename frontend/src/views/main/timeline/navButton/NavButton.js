import React from "react";
import { Center, IconButton } from "@chakra-ui/react";

const ButtonWrapper = (props) => <Center p="0.5rem" {...props} />;

const NavButton = (props) => (
    <ButtonWrapper>
        <IconButton
            {...props}
            fontSize="32px"
            color="primaryColor"
            colorScheme="trasparent"
        >
            {props.children}
        </IconButton>
    </ButtonWrapper>
);

export default NavButton;
