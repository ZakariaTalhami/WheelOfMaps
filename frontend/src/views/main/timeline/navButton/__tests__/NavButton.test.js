import React from "react";
import { render, fireEvent, screen } from "../../../../../utils/TestUtils";
import NavButton from "../NavButton";

test("renders", () => {
    render(<NavButton>Button</NavButton>);
});

test("is Clickable", () => {
    const clickHandler = jest.fn();
    render(
        <NavButton onClick={clickHandler} children="Button">
            Button
        </NavButton>
    );

    expect(clickHandler).not.toHaveBeenCalled();

    fireEvent.click(screen.getByRole("button"));

    expect(clickHandler).toHaveBeenCalled();
});

test("matches snapshot - no selection", () => {
    const { container } = render(<NavButton>Button</NavButton>);

    expect(container).toMatchSnapshot();
});
