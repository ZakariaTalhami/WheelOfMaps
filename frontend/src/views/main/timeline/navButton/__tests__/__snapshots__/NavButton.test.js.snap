// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot - no selection 1`] = `
<div>
  <div
    class="css-ug2oqc"
  >
    <button
      class="chakra-button css-1a8ali6"
      type="button"
    />
  </div>
</div>
`;
